fawltydeps/utils.py

```diff
@@ -22,9 +22,12 @@ def version() -> str:
 
 def dirs_between(parent: Path, child: Path) -> Iterator[Path]:
     """Yield directories between 'parent' and 'child', inclusive."""
-    yield child
-    if child != parent:
-        yield from dirs_between(parent, child.parent)
+    if not child.is_relative_to(parent):
+        raise ValueError(f"{child} is not a child of {parent}")
+    for path in [child, *child.parents]:
+        yield path
+        if path == parent:
+            return
```

Comment:
Alt impl: No special-case, just the empty list when there is no relatives:

```suggestion
    return itertools.takewhile(lambda p: p.is_relative_to(parent), [child, *child.parents])
```

Reply:
Or maybe even 

```
from itertools import takewhile
from operator import methodcaller
# ...
    return takewhile(methodcaller('is_relative_to', parent), [child, *child.parents])
```

not sure if methodcaller is more idiomatic than a lambda in this case.

Reply:
Love it! I committed this with an updated docstring in e7b5d6ec3ee690b588e889be42cf82499f5f6519. Will merge this soon, unless there are more comments.